src/bet_size.rs

```diff
@@ -149,23 +213,45 @@ fn parse_float(s: &str) -> Option<f64> {
     }
 }
 
-fn bet_size_from_str(s: &str, is_raise: bool) -> Result<BetSize, String> {
+fn bet_sizes_from_str(bets_str: &str) -> Result<Vec<BetSize>, String> {
+    let mut bet_sizes = bets_str.split(',').map(str::trim).collect::<Vec<_>>();
+
+    if bet_sizes.last().unwrap().is_empty() {
+        bet_sizes.pop();
+    }
+
+    let mut bets = Vec::new();
+
+    for bet_size in bet_sizes {
+        bets.push(bet_size_from_str(bet_size)?);
+    }
+
+    bets.sort_unstable_by(|l, r| l.partial_cmp(r).unwrap());
+
+    Ok(bets)
+}
+
+fn deserialize_bet_sizes<'de, D>(deserializer: D) -> Result<Vec<BetSize>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s = String::deserialize(deserializer)?;
+    let bet_sizes = bet_sizes_from_str(&s);
+    bet_sizes.map_err(de::Error::custom)
+}
+
+fn bet_size_from_str(s: &str) -> Result<BetSize, String> {
```

Comment:
I think you can just inline this in `impl TryFrom<&str> for BetSize` and replace calls to it with `BetSize::try_from(s)`

Reply:
Done